app/controllers/NotificationController.java

```diff
@@ -47,10 +52,42 @@
         });
     }
 
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> addNotification(Long stageId) {
+        JsonNode json = request().body().asJson();
+        return parseNotification(json).thenApply(notification -> notificationRepository.addNotification(stageId, notification)).
+                thenApplyAsync(dbNotification -> ok(toJson(dbNotification)))
+                .exceptionally(ex -> {
+                    Result res;
+                    switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)